doc/manual/merging.md

```diff
@@ -0,0 +1,117 @@
+# Merging records
+
+When you have a big configuration, the way to split it in several files
+(e.g.: separated by cathegorie or level of abstraction) is the merging operator.
+In nickel this is the `&` operator.
+
+For instence, having a server config and a firewall config, the network config
+could be:
+
+```text
+let
+server = import "server.ncl",
+firewall = import "firewall.ncl",
+in
+server & firewall
+```
+
+In the simple case where both records do not have common fields, it gives a
+record which is the union of both.
+
+If you have to merge records with fields in commons, you will use two new concepts:
+
+- default annotation,
+- overwriting,
+
+## Default annotation
+
+To be able to merge records with commons fields, at least one side field has to
+be annotated as default. This means, if you have two records:
+
+```text
+let left = {
+  firewall.enabled = true,
+  firewall.type = "iptables",
+  firewall.open_ports = [21, 80, 443],
+} in
+let right = {
+  firewall.enabled = false,
+  server.host.options = "TLS",
+} in
+left & right
+```
+
+Like it is, it's impossible to merge and will throw an unmergeable terms error.
+Here, the issue is on the field `firewall.enabled`. What you have to do to fix it
+is to annotate this field as default in one of the records:
+
+```text
+let left = {
+  firewall.enabled | default = true,
+  firewall.type = "iptables",
+  firewall.open_ports = [21, 80, 443],
+} in
+let right = {
+  firewall.enabled = false,
+  server.host.options = "TLS",
+} in
+left & right
+// => {firewall.enabled = false, ...}
+```
+
+The default annotation is generaly to give a default value to a record field.
+So, this value can be changed afterward. Saying it in a more abstract way,
+default indicate a lower priority to a field in case of merging. Saying that,
+If both sides have been annotated `default`, the merge is not possible.
+
+## Overwriting
+
+The overwriting is the concept specifying the behaviour of a merge when you
+overwrite a field on which depends an other one. This  feature is described in
+["#573 [RFC] Merge types and terms syntax proposal"](https://github.com/tweag/nickel/pull/573)
```

Comment:
Don't link to pull requests in the documentation. Don't link to merged RFCs either (they are internal documentation).

Well, let me temper that. You can link to them as further reading for the curious reader (links are good!). But they can't be the introduction, and the understanding of the documentation can't depend on reading them.

You are allowed to copy large bits of them though!